CMake/HighFiveTargetExport.cmake

```diff
@@ -1,48 +1,100 @@
 add_library(HighFive INTERFACE)
-add_library(highfive_deps INTERFACE)
 
 # Public headers
+
 target_include_directories(HighFive INTERFACE
-  "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
-  "$<INSTALL_INTERFACE:include>")
+    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include>")
+
+# Add dependencies to "HighFive" target
+
+target_link_libraries(HighFive INTERFACE highfive_deps)
+target_compile_definitions(HighFive INTERFACE MPI_NO_CPPBIND)  # No c++ bindings
 
+# Ensure we activate at least C++11
 
-# Generate ${PROJECT_NAME}Config.cmake
+if(NOT DEFINED CMAKE_CXX_STANDARD)
+    if(CMAKE_VERSION VERSION_LESS "3.1")
+        message(WARNING "HighFive requires at least c++11. You may need to set CMAKE_CXX_STANDARD.")
+    else()
+        target_compile_features(HighFive INTERFACE cxx_std_11)
+    endif()
+endif()
+
+# Generate HighFiveConfig.cmake
 
 include(CMakePackageConfigHelpers)
-configure_package_config_file(${CMAKE_CURRENT_LIST_DIR}/HighFiveConfig.cmake.in
-  ${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake
-  INSTALL_DESTINATION share/${PROJECT_NAME}/CMake)
+
+configure_package_config_file(
+        ${CMAKE_CURRENT_LIST_DIR}/HighFiveConfig.cmake.in
+        ${PROJECT_BINARY_DIR}/HighFiveConfig.cmake
+    INSTALL_DESTINATION
+        share/HighFive/CMake)
 
 write_basic_package_version_file(
-    ${PROJECT_NAME}ConfigVersion.cmake
-    VERSION ${PROJECT_VERSION}
-    COMPATIBILITY AnyNewerVersion)
+        HighFiveConfigVersion.cmake
+    VERSION
+        ${PROJECT_VERSION}
+    COMPATIBILITY
+        AnyNewerVersion)
+
+install(
+    FILES
+        CMake/HighFiveTargetDeps.cmake
+        ${PROJECT_BINARY_DIR}/HighFiveConfig.cmake
+        ${PROJECT_BINARY_DIR}/HighFiveConfigVersion.cmake
+    DESTINATION
+        share/HighFive/CMake)
+
+# Generate HighFiveTargets.cmake
+
+install(
+    TARGETS
+        HighFive
+    EXPORT
+        HighFiveTargets
+    INCLUDES DESTINATION
+        include)
+
+install(
+    EXPORT
+        HighFiveTargets
+    FILE
+        HighFiveTargets.cmake
+    DESTINATION
+        share/HighFive/CMake)
 
-install(FILES
-    CMake/HighFiveTargetDeps.cmake
-    ${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake
-    ${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake
-  DESTINATION share/${PROJECT_NAME}/CMake)
+export(
+    EXPORT
+        HighFiveTargets
+    FILE
+        "${PROJECT_BINARY_DIR}/HighFiveTargets.cmake")
 
+# Install static target "highfive_deps" if requested
+# If not installed it is constructed when the user calls "find_package(HighFive)"
 
-# Generate ${PROJECT_NAME}Targets.cmake;
-# Provides IMPORTED targets when using this project from build/install trees.
+if (HIGHFIVE_USE_INSTALL_DEPS)
```

Comment:
Neat the idea to export as a separate target file. However, please always export the highfive_deps, HIGHFIVE_USE_INSTALL_DEPS should only be the default for the end project, but it may still override it and then this info must be avail.

Reply:
I think there should still be a way to switch off the creation of the 'static' target. For example when shipping the package from conda-forge I really think that should be not such target shipped. Do you have a suggestion for an option name?

Reply:
I have proposed `HIGHFIVE_INSTALL_STATIC`